march_hardware/include/march_hardware/imotioncube/motor_controller.h

```diff
@@ -0,0 +1,39 @@
+// Copyright 2019 Project March.
+
+#include "actuation_mode.h"
+#include "motor_controller_state.h"
+#include <string>
+
+namespace march
+{
+class MotorController
+{
+public:
+  virtual double getAngleRadAbsolute() = 0;
+  virtual double getAngleRadIncremental() = 0;
+  virtual double getVelocityRadAbsolute() = 0;
+  virtual double getVelocityRadIncremental() = 0;
+  virtual int16_t getTorque() = 0;
+  virtual MotorControllerStates getStates() = 0;
+
+  virtual ActuationMode getActuationMode() const = 0;
+  virtual uint16_t getSlaveIndex() const = 0;
+
+  virtual float getMotorCurrent() = 0;
+  virtual float getMotorControllerVoltage() = 0;
+  virtual float getMotorVoltage() = 0;
+  virtual bool getIncrementalMorePrecise() const = 0;
+
+  virtual void actuateRad(double target_rad) = 0;
+  virtual void actuateTorque(int16_t target_torque) = 0;
+
+  virtual void goToOperationEnabled() = 0;
+
+  virtual bool initialize(int cycle_time) = 0;
```

Comment:
What if my motor controller does not need a cycle time, but it does need something else for initialization. That would require adding more parameters and therefore breaking the contract.

Reply:
I think you could alternatively pass something like that to the constructor as well, but I guess that wouldn't be very need since you need it nowhere else. I think I am going to :ostrich: this problem since it is highly dependent on how exactly the startup of another motor controller would work, which is a bit difficult to predict. I will see if I get useful enough insight from the ODrive and the Ingenia to make general statements about this. Otherwise, I think I will leave it up to future generations to redesign this a bit.